modular/blocksyncer/blocksyncer_indexer.go

```diff
@@ -1,551 +0,0 @@
-		if err != nil {
-			return err
-		}
-	}
-
-	allEvents := make([][]TxHashEvent, 0)
-	allEvents = append(allEvents, bucketEvent)
-	allEvents = append(allEvents, groupEvent)
-	allEvents = append(allEvents, objectEvent)
-	allEvents = append(allEvents, permissionEvent)
-	allEvents = append(allEvents, spEvent)
-	allEvents = append(allEvents, prefixEvent)
-	allEvents = append(allEvents, VirtualGroupEvent)
-	allEvents = append(allEvents, exitEvent)
-	allEvents = append(allEvents, objectIDEvent)
-
-	return i.concurrenceHandleEvent(ctx, block, allEvents)
-}
-
-// HandleGenesis accepts a GenesisDoc and calls all the registered genesis handlers in the order in which they have been registered.
-func (i *Impl) HandleGenesis(genesisDoc *tmtypes.GenesisDoc, appState map[string]json.RawMessage) error {
-	return nil
-}
-
-// HandleBlock accepts block and calls the block handlers.
-func (i *Impl) HandleBlock(block *coretypes.ResultBlock, events *coretypes.ResultBlockResults, txs []*types.Tx, getTmcValidators modules.GetTmcValidators) {
-	for _, module := range i.Modules {
-		if blockModule, ok := module.(modules.BlockModule); ok {
-			err := blockModule.HandleBlock(block, events, txs, getTmcValidators)
-			if err != nil {
-				log.Errorw("error while handling block", "module", module.Name(), "height", block.Block.Height, "err", err)
-			}
-		}
-	}
-}
-
-// HandleTx accepts the transaction and calls the tx handlers.
-func (i *Impl) HandleTx(tx *types.Tx) {
-	log.Info("HandleTx")
-}
-
-// HandleMessage accepts the transaction and handles messages contained inside the transaction.
-func (i *Impl) HandleMessage(block *coretypes.ResultBlock, index int, msg sdk.Msg, tx *types.Tx) {
-	log.Info("HandleMessage")
-}
-
-// Processed tells whether the current Indexer has already processed the given height of Block
-// An error is returned if the operation fails.
-func (i *Impl) Processed(ctx context.Context, height uint64) (bool, error) {
-	ep, err := i.DB.GetEpoch(context.Background())
-	if err != nil {
-		return false, err
-	}
-	// log.Infof("epoch height:%d, cur height: %d", ep.BlockHeight, height)
-	if ep.BlockHeight > int64(height) {
-		heightKey := fmt.Sprintf("%s-%d", i.GetServiceName(), height)
-		blockMap.Delete(heightKey)
-		eventMap.Delete(heightKey)
-		txMap.Delete(heightKey)
-	}
-	return ep.BlockHeight > int64(height), nil
-}
-
-// GetBlockRecordNum returns total number of blocks stored in database.
-func (i *Impl) GetBlockRecordNum(_ context.Context) int64 {
-	return 1
-}
-
-// GetLastBlockRecordHeight returns the last block height stored inside the database
-func (i *Impl) GetLastBlockRecordHeight(ctx context.Context) (uint64, error) {
-	var lastBlockRecordHeight uint64
-	currentEpoch, err := i.DB.GetEpoch(ctx)
-	if err == nil {
-		lastBlockRecordHeight = 0
-	} else {
-		lastBlockRecordHeight = uint64(currentEpoch.BlockHeight)
-	}
-	return lastBlockRecordHeight, err
-}
-
-func (i *Impl) GetLatestBlockHeight() *atomic.Value {
```

Comment:
why don't we return int64 directly, so the caller doesn't have to change by themselves.

And try to add a function `SetLatestBlockHeight()`  instead of `Cast(b.parserCtx.Indexer).GetLatestBlockHeight().Store(int64(latestBlockHeight))`

Reply:
Good advice. I'll change it